clarity_ext/service/artifact_service.py

```diff
@@ -43,18 +43,18 @@ def all_aliquot_pairs(self):
         """
         Returns all aliquots in a step as an artifact pair (input/output)
         """
-        pairs = self.step_repository.all_artifacts()
-        aliquots_only = filter(lambda pair: isinstance(pair[0], Aliquot) and
-                               isinstance(pair[1], Aliquot), pairs)
+        pairs = self.all_artifacts()
```

Comment:
Was it intentional to do `self.all_artifacts` rather than `self.step_repository.all_artifacts`?

Reply:
Oh, yes. This change do not really belongs here, but it fixes a bug that have irritated me for some time. "step.step_repository.all_artifacts()" always fetch values from the api, whereas "self.all_artifacts()" fetches from an internal cache if able. So if in a script there happen to be two for-loops, looping over artifacts, and if artifacts are updated in both loops, the "step_repository.all_artifacts()" fails, because the first update is overwritten. Note that most of the methods in artifact service uses the internal cache, but this does not until now.

Reply:
I was surprised however that this change was in this PR. Sorry for that.

Reply:
No problem! Sounds excellent :)